### PR TITLE
fix overscroll spacer in safari

### PR DESCRIPTION
**Pull request by ishpaul777**
### Fixed Issues

$ https://github.com/Expensify/App/issues/23422
PROPOSAL: https://github.com/Expensify/App/issues/23422#issuecomment-1666890994

### Tests

- [x] Verify that no errors appear in the JS console

On Safari IOS and IOS Native app:
1. Navigate to preferences page
2. Try to overscroll Down, verify no green background until scrolled 50% of Scrollable view
3. Try to overscroll Upward, verify no illustrated Header color until scrolled 50% of Scrollable view
4. Navigate to Workspace list page
5. Try to overscroll Down, verify no green background until scrolled 50% of Scrollable view
6.  Try to overscroll Upward, verify no illustrated Header color until scrolled 50% of Scrollable view

### Offline tests

On Safari IOS and IOS Native app:
1. Navigate to preferences page
2. Try to overscroll Down, verify no green background until scrolled 50% of Scrollable view
3. Try to overscroll Upward, verify no illustrated Header color until scrolled 50% of Scrollable view
4. Navigate to Workspace list page
5. Try to overscroll Down, verify no green background until scrolled 50% of Scrollable view
6.  Try to overscroll Upward, verify no illustrated Header color until scrolled 50% of Scrollable view

### QA Steps

- [x] Verify that no errors appear in the JS console

On Safari IOS and IOS Native app:
1. Navigate to preferences page
2. Try to overscroll Down, verify no green background until scrolled 50% of Scrollable view
3. Try to overscroll Upward, verify no illustrated Header color until scrolled 50% of Scrollable view
4. Navigate to Workspace list page
5. Try to overscroll Down, verify no green background until scrolled 50% of Scrollable view
6.  Try to overscroll Upward, verify no illustrated Header color until scrolled 50% of Scrollable view

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web Safari</summary>

https://github.com/Expensify/App/assets/104348397/0d82caa4-9802-489f-89d5-5edb4388e4d6

</details>

<details>
<summary>Web Chrome</summary>

https://github.com/Expensify/App/assets/104348397/1576201b-93d4-43d8-8ddb-8f78515a9f31

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/104348397/018423e6-ddc7-41d2-af0c-b591712ea499

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/104348397/e4ba8935-a319-40a8-93a3-717821db66af

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/104348397/fd94b39a-580a-46cc-87ca-3f56dd564ea0

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/104348397/eec8b34b-dec9-47aa-b681-c4c9df8c7682

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/104348397/f897072f-f7d6-4c60-a5d7-5b9a50d5b675

</details>

### CODE REVIEW

**Comment by ishpaul777** [comment]
done

**Review by 0xmiroslav — APPROVED** [review]
LGTM! (still workaround though)

**Comment by ishpaul777** [comment]
@roryabraham  Please Hold youre review I fould a bug, pushing fix in few minutes

File: src/components/IllustratedHeaderPageLayout.js
Comment (ishpaul777) [inline src/components/IllustratedHeaderPageLayout.js:58]: Made changes specifc to safari, on Android when there is only one workspace there is dual background visible.

File: src/components/IllustratedHeaderPageLayout.js
Comment (ishpaul777) [inline src/components/IllustratedHeaderPageLayout.js:61]: for the large screen when workspace count is less upper background was visible, On mac safari, overscrolling is allowed very little so i guess this is fine

**Comment by ishpaul777** [comment]
@0xmiroslav @roryabraham Bumping for review

**Comment by ishpaul777** [comment]
I apology i changed few things commit was not signed, reverted and forgot to push again, will made change again give me few mins.

**Review by 0xmiroslav — APPROVED** [review]
LGTM!

I think @roryabraham will be busy this week (attending in RNEU conference)
Hopefully reviewed & merged next Monday

**Review by roryabraham — DISMISSED** [review]
I was still able to reproduce this in Safari with a vigorous scroll, but it's much better than it was. 

https://github.com/Expensify/App/assets/47436092/144e8e0c-88df-453e-bb68-b1d52de6b5de

We also can't really shift it so that the blue background takes up more than half of the screen because then we'll just introduce the inverse problem on the bottom.

So with this PR you can still see the bug if you intentionally over-scroll beyond what you would normally do. The solution isn't perfect but it's a workaround to a safari bug that makes it better for most users most of the time, and I'll take that in this case because we can't fix it upstream in Safari.

**Comment by roryabraham** [comment]
This is not urgent, so I'd like to wait until after we do one more production release before merging this. Because of the merge freeze, if we're not careful we'll have a GIANT release checklist on our hands if we merge everything the first day the freeze is lifted.

This delay should not count against the #urgency bonus. I think the PR was made and approved by C+ within the time limit to get the #urgency bonus so unless something changes I think this PR should receive the #urgency bonus

**Comment by roryabraham** [comment]
Thanks for your patience here @ishpaul777. Can you please resolve conflicts so we can get this merged?

**Comment by ishpaul777** [comment]
@0xmiroslav android app is build, tested on all other platforms, also How can i test lounge access page?

**Comment by ishpaul777** [comment]
Lounge access page has BG same as appBG so fix has effect on the page. Android app is still building, sorrry for delay.

https://github.com/Expensify/App/assets/104348397/a6c9eeaa-401a-4118-b240-3399337ce193

**Comment by ishpaul777** [comment]
Merge changes tests well on all platforms @0xmiroslav

**Review by 0xmiroslav — APPROVED** [review]
Tests well for me too

@roryabraham all good to merge!

**Review by roryabraham — APPROVED** [review]

**Pull request merged**
